core/src/main/java/jenkins/mvn/GlobalSettingsProvider.java

```diff
@@ -21,13 +26,45 @@
 public abstract class GlobalSettingsProvider extends AbstractDescribableImpl<GlobalSettingsProvider> implements ExtensionPoint {
 
     /**
+     * <p>
+     *     Configure maven launcher argument list with adequate settings path.
+     * </p>
+     * <p>Implementations should
+     * <ul>Be aware that this method might get called multiple times during a build.</ul>
+     * <ul>Implement this method. This class provides a default implementation throwing an {@link UnsupportedOperationException}
+     * so that implementations have time to adapt.</ul>
+     * </p>
+     *
+     * @param run       the build / run to provide the settings for
+     * @param workspace the workspace in which the build / run takes place
+     * @param listener the listener of this given build / run
+     * @return the filepath to the provided file. <code>null</code> if no settings will be provided.
+     */
+    @CheckForNull
+    public FilePath supplySettings(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnull TaskListener listener) throws IOException, InterruptedException {
+        throw new AbstractMethodError("Class " + getClass() + " must override the new supplySettings overload");
```

Comment:
I would rather call to the old `supplySettings` method if that method is overridden && the run is AbstractBuild. Otherwise I would print warning and return null. In such case new API won't blow up if somebody invokes it for non-updated providers